daemon/update_linux.go

```diff
@@ -15,7 +15,9 @@ func toContainerdResources(resources container.Resources) libcontainerd.Resource
 	r.CpuQuota = uint64(resources.CPUQuota)
 	r.CpusetCpus = resources.CpusetCpus
 	r.CpusetMems = resources.CpusetMems
-	r.MemoryLimit = uint64(resources.Memory)
+	if resources.Memory != 0 {
+		r.MemoryLimit = uint64(resources.Memory)
```

Comment:
This `if` condition is not needed.

So we're hacking like:
uint64(-1) = 18446744073709551615
int64(18446744073709551615) = -1

Is everyone comfortable with that? I prefer we change value type to explicitly say we allow negative value here.